src/app/user-sign-up/register/register.component.ts

```diff
@@ -9,37 +12,99 @@ import { Router } from '@angular/router';
 })
 export class RegisterComponent implements OnInit {
   registerForm!: FormGroup;
+  /**
+   * Constructor
+   */
+  constructor(private addUsers: UserService, private router: Router) { };
 
-  // Constructor
-  constructor(private router: Router){};
-
-  // ngOnInit method
-  ngOnInit(): void{
+  /**
+   * ngOnInit method
+   */
+  ngOnInit(): void {
     this.registerForm = new FormGroup({
-        name: new FormControl('', [Validators.required]),
-        email : new FormControl('', [Validators.required , Validators.email]),
-        password : new FormControl('',
-                          [Validators.required, 
-                            Validators.pattern(/^(?=.*[A-Z])(?=.*[a-z])(?=.*[0-9])(?=.*[!@#\$%\^&\*])(?=.{8,})/),
-                            Validators.minLength(8)
-                          ])
+      name: new FormControl('', [Validators.required]),
+      email: new FormControl('', [Validators.required, Validators.pattern("^[a-z0-9._%+-]+@[a-z0-9.-]+\\.[a-z]{2,4}$")]),
+      password: new FormControl('',
+        [Validators.required,
+        Validators.pattern(/^(?=.*[A-Z])(?=.*[a-z])(?=.*[0-9])(?=.*[!@#\$%\^&\*])(?=.{8,})/),
+        Validators.minLength(8)
+        ])
     })
   }
 
-  // getter methods
-  get name(){
+  /**
+   * name method are used to get name
+   * @retun name
+   *  */
+  get name() {
     return this.registerForm.get('name');
   }
-  get email(){
+
+  /**
+   * email method are used to get email
+   * @retun email
+   *  */
+  get email() {
     return this.registerForm.get('email');
   }
-  get password(){
+
+  /**
+   * password method are used to get password
+   * @retun password
+   *  */
+  get password() {
     return this.registerForm.get('password');
   }
 
-  // Form submit method
-  onSubmit(): void{
+  /**
+   * addUser method are used to subscribe data
+   * @param new_user : UserRegister
```

Comment:
new_user -> newUser
also the description of params

Reply:
> new_user -> newUser also the description of params

Ok Dii